packages/manager/src/features/CloudPulse/Alerts/Utils/utils.ts

```diff
@@ -118,3 +126,59 @@ export const getChipLabels = (
     };
   }
 };
+
+export interface ProcessedCriteria {
+  aggregationType: string;
+  label: string;
+  operator: string;
+  threshold: number;
+  unit: string;
+}
+
+export const processMetricCriteria = (
+  criterias: AlertDefinitionMetricCriteria[]
+): ProcessedCriteria[] => {
+  return criterias
+    .map((criteria) => {
+      const { aggregate_function, label, operator, threshold, unit } = criteria;
+      return {
+        aggregationType: aggregationTypeMap[aggregate_function],
+        label,
+        operator: metricOperatorTypeMap[operator],
+        threshold,
+        unit,
+      };
+    })
+    .reduce<ProcessedCriteria[]>((previousValue, currentValue) => {
+      previousValue.push(currentValue);
+      return previousValue;
+    }, []);
+};
+
+export const filterAlertsByStatusAndType = (
+  alerts: Alert[] | undefined,
+  searchText: string,
+  selectedType: string | undefined
+): Alert[] => {
+  return (
+    alerts?.filter((alert) => {
+      return (
+        alert.status === 'enabled' &&
+        (!selectedType || alert.type === selectedType) &&
+        (!searchText ||
+          alert.label.toLowerCase().includes(searchText.toLowerCase()))
+      );
+    }) ?? []
+  );
+};
```

Comment:
```suggestion
export const filterAlertsByStatusAndType = (
  alerts: Alert[] | undefined,
  searchText: string,
  selectedType: string | undefined
): Alert[] => {
  return (
    alerts?.filter(({ label, status, type }) => {
      return (
        status === 'enabled' &&
        (!selectedType || type === selectedType) &&
        (!searchText || label.toLowerCase().includes(searchText.toLowerCase()))
      );
    }) ?? []
  );
};
```

Lets desctructure the alert inside filter and use for good readability